addons/necropost-highlighter/userscript.js

```diff
@@ -0,0 +1,333 @@
+/**
+ * @author RT_Borg
+ *
+ * Uses topic id as a proxy for age, which doesn't require loading other pages
+ * to read actual dates.
+ */
+
+export default async function ({ addon, global, console, msg }) {
+  // Default is a little over a month's worth of topics. Reasonably stable through 2022.
+  // There's a lot of tolerance. Half or twice as many topics filter pretty much the same.
+  const defaultTopicInterval = 7500;
+  const defaultForums = [
+    "New Scratchers",
+    "Help with Scripts",
+    "Questions about Scratch",
+    "Bugs and Glitches",
+    "Project Ideas",
+  ];
+  // Forums beyond these seem very poor options,
+  // as many posts live long or are appropriately revived
+  const customForumSettingToForum = {
+    applyToNewScratchers: "New Scratchers",
+    applyToHelpWithScripts: "Help with Scripts",
+    applyToQuestionsAboutScratch: "Questions about Scratch",
+    applyToBugsAndGlitches: "Bugs and Glitches",
+    applyToProjectIdeas: "Project Ideas",
+    applyToRequests: "Requests",
+    applyToSuggestions: "Suggestions",
+    applyToOpenSourceProjects: "Open Source Projects",
+  };
+  const searchResultsPageName = "Search Results"; // recent posts and unanswered posts
+  // mobile urls look like: https://scratch.mit.edu/discuss/m/6/*
+  const isMobileSite = /^\/discuss\/m\//.test(location.pathname);
+  // tableBodyNode will be null on mobile and non-table search page:
+  // https://scratch.mit.edu/discuss/search/?action=show_user&show_as=posts
+  const tableBodyNode = document.querySelector("tbody"); // null on mobile
+
+  if (!tableBodyNode && !isMobileSite) {
+    // Standard site page without a table. Don't run on this page.
+    return;
+  }
+
+  const forumOrSearchPageName = await determineForumOrSearchPageName();
+
+  /**
+   * An array of topic
+   * Each topic is:
+   *    - topicId      int, used to compare topic "age" relative to other posts on the page
+   *    - topicCell    Node that will be highlighted if too old
+   *    - forum        String, usually the forum being listed. Comes from the row on a search page
+   *    - restoreCell  Only defined if the cell is modified. A clone of the initial state
+   *                   of the cell Node, to revert for disable addon reload and settings changes
+   */
+  let topics = [];
+
+  highlightNecropostsIfOnSelectedForumPage();
+
+  addon.settings.addEventListener("change", onAddonSettingsChange);
+  addon.self.addEventListener("disabled", onDisabled);
+  addon.self.addEventListener("reenabled", onReenabled);
+
+  // Support for infinite scrolling (only available on Standard not Mobile)
```

Comment:
Do we need to create a `MutationObserver` on mobile pages, if Infinite Scrolling doesn't work?

Reply:
I think you're suggesting to declare tableBodyMutationObserver here, but assign it inside the !isMobileSite guard of attach (and free it in detach). I'll make a commit for this.